modules/configuration/src/main/java/org/apache/ignite/configuration/internal/selector/BaseSelectors.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.configuration.internal.selector;
+
+import java.lang.invoke.MethodHandle;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.ignite.configuration.internal.DynamicConfiguration;
+import org.apache.ignite.configuration.internal.property.Modifier;
+
+/**
+ * Base selector holder.
+ */
+public class BaseSelectors {
+    /** Map from string representation of selector to {@link SelectorHolder}. */
+    private static final Map<String, SelectorHolder> selectors = new HashMap<>();
+
+    /**
+     * Get selector from selectors map by key.
+     *
+     * Valid formats for selector key:
+     * <ul>
+     *     <li>root.inner.option.field in case of static config field</li>
+     *     <li>root.inner.named[name].field in case of dynamic (named) config field</li>
+     * </ul>
+     *
+     * @param name Selector name.
+     * @return Selector.
+     */
+    public static <A extends DynamicConfiguration<?, ?, ?>, B extends Modifier<C, D, E>, C, D, E> Selector<A, B, C, D, E> find(String name) {
+        String[] splitten = name.split("\\.");
```

Comment:
Naming is weird. First of all, no one uses the word "splitten", second - we can name it, like, "names" of "propNames", you know?

Reply:
`true`